src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -557,11 +610,31 @@
                                                extensions)]
     (write-cert-to-inventory! hostcert (:cert-inventory ca-settings))
     (utils/key->pem! public-key (:hostpubkey settings))
-    (utils/key->pem! private-key (:hostprivkey settings))
+    (utils/key->pem! private-key
+     (create-file-with-perms (:hostprivkey settings) p-key-perms))
     (utils/cert->pem! hostcert (:hostcert settings))
     (utils/cert->pem! hostcert
                       (path-to-cert (:signeddir ca-settings) certname))))
 
+(schema/defn ^:always-validate
+  ensure-master-file-perms!
+  [settings :- MasterSettings]
+  (let [p-key-dir (:privatekeydir settings)
+        cur-dir-perms (get-file-perms p-key-dir)]
+    (when-not (= p-key-dir-perms cur-dir-perms)
+      (set-file-perms p-key-dir p-key-dir-perms)
+      (log/warnf (str "The private key directory at '%s' had permissions set as "
+                      "'%s' and should be set to '%s' this has been corrected")
+                 p-key-dir cur-dir-perms p-key-dir-perms))
+    (doseq [key-file (fs/find-files p-key-dir #".*pem$")]
+      (let [key-path (.getPath key-file)
+            cur-key-perms (get-file-perms key-path)]
+        (when-not (= p-key-perms cur-key-perms)
+          (set-file-perms key-path p-key-perms)
+          (log/warnf (str "The private key located at '%s' had permissions set as "
+                          "'%s' and should be set to '%s' this has been fixed.")
```

Comment:
I guess practically it will be hard for any of this code to be hit since the underlying Ruby initialization code will probably correct the permissions anyway before we get here.  I'm still good with this code being here as a sanity check anyway, though - i.e., not just trusting the Ruby layer to *magically* fix things up.